### DISCUSSION TITLE
Is there a way to use h1 as `title:` in metadata if `title:` isn't specified?

### DISCUSSION THREAD

**Discussion by the-solipsist** [quarto-dev/quarto-cli#6757] [Q&A] [answered]
I've got 50+ qmd files with `#` title, and that same string copied in the metadata with `title:`.  

When I render the files as html, the title gets duplicated, with 2 `<h1>` headings.  

If I remove `title:`, then the title-metadata block doesn't get rendered correctly.  I've tried `--shift-heading-by=-1`, but that makes `##` `<h2>` headings into `#` `<h1>` headings.  I can manually remove the `#` heading for each of the files. But I'm wondering if there's any way to make quarto treat `<h1>` headings as titles.

**Comment by mcanouil** [comment]
Would you be able to share the code ? I am not fully sure to understand your case.

**Comment by the-solipsist** [comment]
I have a file like this:

```qmd
---
title: "Grand Title"
author: "Pranesh Prakash"
date: 2011-01-25
standalone: true
shift-heading-level-by: 0
---

# Grand Title

Lorem. 

## Heading 2

Lorem ipsum.
````
The `<h1>` title appears twice:

![image](https://github.com/quarto-dev/quarto-cli/assets/3080824/b748d11b-f720-4666-af9e-cb7fa1dfeb31)

```qmd
---
title: "Grand Title"
author: "Pranesh Prakash"
date: 2011-01-25
standalone: true
shift-heading-level-by: -1
---

# Grand Title

Lorem. 

## Heading 2

Lorem ipsum.

```
If I use `shift-heading-level-by: -1`, then the title issue gets fixed, but "Heading 2" becomes an `<h1>`, which I don't want.

![image](https://github.com/quarto-dev/quarto-cli/assets/3080824/409c5eca-3fea-4878-b8c4-798468ac397f)


```qmd
---
author: "Pranesh Prakash"
date: 2011-01-25
standalone: true
shift-heading-level-by: 0
---

# Grand Title

Lorem. 

## Heading 2

Lorem ipsum.
```
If I remove `title:` from the metadata and _don't_ shift the heading levels, then the metadata block disappears:

![image](https://github.com/quarto-dev/quarto-cli/assets/3080824/faf98e9e-015b-4b08-bb20-502b13167def)

```qmd
---
title: "Grand Title"
author: "Pranesh Prakash"
date: 2011-01-25
standalone: true
shift-heading-level-by: 0
---

Lorem. 

## Heading 2

Lorem ipsum.

```
What I'd like is ideally this _while_ keeping `# Grand Title`:

![image](https://github.com/quarto-dev/quarto-cli/assets/3080824/5cb4fbb4-7e32-46b4-843a-8877c2a13cf2)

**Reply by mcanouil** [reply]
Why do you have Quarto document with h1 title and title set if you don't want that?
This is the standard behaviour of Quarto.
What you could do is write a small Lua filter to remove h1 title that do not have the title class.

**Comment by the-solipsist** [comment, marked as answer]
If anyone reading this wants to know how to get this done, either: 
1. delete all instances of `# title`, or
2. Use this lua filter (adapted from @tarleb's lua filter [here](https://stackoverflow.com/a/56005271))

```lua
local title

-- Set title from level 1 header, or
-- discard level 1 header if title is already set.
function promote_header_title (header)

  if header.level >= 2 then
    return header
  end

  if not title then
    title = header.content
    return {}
  end

  local msg = '[WARNING] title already set; discarding header "%s"\n'
  io.stderr:write(msg:format(pandoc.utils.stringify(header)))
  return {}
end

return {
  {Meta = function (meta) title = meta.title end}, -- init title
  {Header = promote_header_title},
  {Meta = function (meta) meta.title = title; return meta end}, -- set title
}
```
